### DISCUSSION TITLE
How to request

### DISCUSSION THREAD

**Discussion by offpiste-reese** [Budibase/budibase#5428] [Help]
I have the following scenario I would like to implement in Budibase:
User logs into Lime Survey via the Lime REST API and gets a session id -> I've implemented this

After login another REST call to list the surveys this user has access to, display the list -> I can make the call, but how to display the list?  If I use a data provider, how do I change parameters on the underlying REST query?  Additionally, when I hook up a Table to the data provider, the table just shows the JSON text.  In the Column dropdown of the datasource it only shows the top level array name, not the individual columns.  If I execute the query in the button action rather than using a datasource, how do I display multiple rows of data?

Then I have multiple REST calls after that return rows of data all based on data selections the user makes from  previeous REST data.  Perhaps I haven't found the right documentation, but I don't see how to do this.

**Comment by melohagan** [comment]
Hey @offpiste-reese 

It sounds like you need to use a transformer in your REST query to return a valid row schema:
https://docs.budibase.com/docs/transformers

As for changing the parameters of a REST query from a data provider, this should help: https://docs.budibase.com/docs/rest-bindings#data-provider-applications-provider
